src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -67,7 +67,7 @@ std::vector<std::shared_ptr<AbstractExpression>> StoredTableNode::column_express
     const auto table = Hyrise::get().storage_manager.get_table(table_name);
 
     // Build `_expression` with respect to the `_pruned_column_ids`
-    _column_expressions.emplace(table->column_count() - _pruned_column_ids.size());
+    _column_expressions.emplace(table->column_count() - _pruned_column_ids.size());  // <- does this make sense?
```

Comment:
I don't get this: Why do we are create an `AbstractExpression` from an Int value? Looks strange.

Reply:
We don't. `_column_expressions` is an `optional<vector<...>>`, so we call `optional::emplace`, which in turn calls the `size_type` constructor of `vector<...>`. Smells a lot though and should be `_column_expressions = std::vector<std::shared_ptr<AbstractExpression>>(table->column_count() - _pruned_column_ids.size());`